lib/elasticity/aws_request.rb

```diff
@@ -50,68 +50,28 @@ def host
       "elasticmapreduce.#{@region}.amazonaws.com"
     end
 
-    def credential_scope
-      "#{@timestamp.strftime('%Y%m%d')}/#{@region}/#{SERVICE_NAME}/aws4_request"
-    end
-
-    # Task 1: Create a Canonical Request For Signature Version 4
-    #   http://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
-    def canonical_request
-      [
-        'POST',
-        '/',
-        '',
-        'content-type:application/x-amz-json-1.1',
-        "host:#{host}",
-        "user-agent:elasticity/#{Elasticity::VERSION}",
-        "x-amz-content-sha256:#{Digest::SHA256.hexdigest(payload)}",
-        "x-amz-date:#{@timestamp.strftime('%Y%m%dT%H%M%SZ')}",
-        "x-amz-target:ElasticMapReduce.#{@operation}",
-        '',
-        'content-type;host;user-agent;x-amz-content-sha256;x-amz-date;x-amz-target',
-        Digest::SHA256.hexdigest(payload)
-      ].join("\n")
-    end
-
-    # Task 2: Create a String to Sign for Signature Version 4
-    #   http://docs.aws.amazon.com/general/latest/gr/sigv4-create-string-to-sign.html
-    def string_to_sign
-      [
-        'AWS4-HMAC-SHA256',
-        @timestamp.strftime('%Y%m%dT%H%M%SZ'),
-        credential_scope,
-        Digest::SHA256.hexdigest(canonical_request)
-      ].join("\n")
-    end
-
-    # Task 3: Calculate the AWS Signature Version 4
-    #   http://docs.aws.amazon.com/general/latest/gr/sigv4-calculate-signature.html
-    def aws_v4_signature
-      date = OpenSSL::HMAC.digest('sha256', 'AWS4' + @secret_key, @timestamp.strftime('%Y%m%d'))
-      region = OpenSSL::HMAC.digest('sha256', date, @region)
-      service = OpenSSL::HMAC.digest('sha256', region, SERVICE_NAME)
-      signing_key = OpenSSL::HMAC.digest('sha256', service, 'aws4_request')
-
-      OpenSSL::HMAC.hexdigest('sha256', signing_key, string_to_sign)
+    def headers_to_sign
+      headers = {
+        'Content-Type' => 'application/x-amz-json-1.1',
+        'Host' => host,
+        'User-Agent' => "elasticity/#{Elasticity::VERSION}",
+        'X-Amz-Date' => @timestamp.strftime('%Y%m%dT%H%M%SZ'),
+        'X-Amz-Target' => "ElasticMapReduce.#{@operation}",
+      }
+      if !@session_token.nil?
+        headers['X-Amz-Security-Token'] = @session_token
+      end
+      headers
     end
 
     def submit(ruby_params)
-      if ruby_params.key?(:release_label)
```

Comment:
The `if..else` here, and below, seems intended to switch between a Sigv4 and Sigv2 signing process depending on the EMR AMI version (e.g., `:release_label`)

Sigv2 is now deprecated, and I observed that Sigv4 worked for requests that would previously have used the Sigv2 code path.

Once support for Sigv2 ends, this would bite us eventually anyway.

Reply:
@rbart-placed apparently I'm still watching this repo since it's public, so got an email on this PR. Just wanted to say hi to you and everyone! Hope everyone is doing well and safe ;)

Reply:
Definitely did a double-take on this email 😂

Reply:
Hey!!! Glad to hear from you, hope all is well on your end too 😀 

So how about approving the PR while you're here? 😉